src/elements/content-sidebar/stories/tests/MetadataSidebarRedesign-visual.stories.tsx

```diff
@@ -54,64 +86,131 @@ export default {
+export const AddingNewMetadataTemplate: StoryObj<typeof MetadataSidebarRedesign> = {
     play: async ({ canvasElement }) => {
         const canvas = within(canvasElement);
-        const heading = await canvas.findByRole('heading', { name: 'Metadata' });
-        expect(heading).toBeInTheDocument();
 
-        const addTemplateButton = await canvas.findByRole('button', { name: 'Add template' });
+        const addTemplateButton = await canvas.findByRole('button', { name: 'Add template' }, { timeout: 5000 });
         expect(addTemplateButton).toBeInTheDocument();
         await userEvent.click(addTemplateButton);
 
-        const customMetadataOption = canvas.getByRole('option', { name: 'Custom Metadata' });
+        const customMetadataOption = canvas.getByRole('option', { name: 'Virus Scan' });
         expect(customMetadataOption).toBeInTheDocument();
         await userEvent.click(customMetadataOption);
 
-        const templateHeader = await canvas.findByRole('heading', { name: 'Custom Metadata' });
+        const templateHeader = await canvas.findByRole('heading', { name: 'Virus Scan' });
         expect(templateHeader).toBeInTheDocument();
     },
 };
 
-export const MetadataInstanceEditorWithTemplate: StoryObj<typeof MetadataSidebarRedesign> = {
+export const UnsavedChangesModalWhenChoosingDifferentTemplate: StoryObj<typeof MetadataSidebarRedesign> = {
+    args: {
+        fileId: '416047501580',
+        metadataSidebarProps: defaultMetadataSidebarProps,
+    },
     play: async ({ canvasElement }) => {
         const canvas = within(canvasElement);
-        const heading = await canvas.findByRole('heading', { name: 'Metadata' });
-        expect(heading).toBeInTheDocument();
 
-        const addTemplateButton = await canvas.findByRole('button', { name: 'Add template' });
+        const addTemplateButton = await canvas.findByRole('button', { name: 'Add template' }, { timeout: 5000 });
         expect(addTemplateButton).toBeInTheDocument();
         await userEvent.click(addTemplateButton);
 
-        const customMetadataOption = canvas.getByRole('option', { name: 'My Template' });
+        const customMetadataOption = canvas.getByRole('option', { name: 'Virus Scan' });
         expect(customMetadataOption).toBeInTheDocument();
         await userEvent.click(customMetadataOption);
 
-        const templateHeader = await canvas.findByRole('heading', { name: 'My Template' });
-        expect(templateHeader).toBeInTheDocument();
+        await userEvent.click(addTemplateButton);
+        const myTemplateOption = canvas.getByRole('option', { name: 'My Template' });
+        expect(myTemplateOption).toBeInTheDocument();
+        await userEvent.click(myTemplateOption);
+
+        const unsavedChangesModal = await screen.findByRole(
+            'heading',
+            { level: 2, name: 'Unsaved Changes' },
+            { timeout: 5000 },
+        );
+        expect(unsavedChangesModal).toBeInTheDocument();
```

Comment:
If I understand the logic of this test correctly, is the requirement to enable editing only one Template Instance at a time? If so, could you please point to this requirement piece?

Reply:
Yes, it was decided to show only the currently edited template instance, when in editing mode.
Let's discuss in more details on Slack.